### PR TITLE
python3Packages.fastapi: fix build

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
https://github.com/tiangolo/fastapi/pull/2335 was merged.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
> This will be available in FastAPI 0.65.0, released in a couple of hours.

Let's give the author a few hours here.

File: pkgs/development/python-modules/json-logging/default.nix [hunk 1]
```diff
@@ -29,11 +30,12 @@ buildPythonPackage rec {
...
-  # - Quart is not packaged for Nixpkgs.
-  # - FastAPI is broken, see #112701 and tiangolo/fastapi#2335.
-  checkInputs = [ wheel flask /*quart*/ sanic /*fastapi*/ uvicorn requests pytestCheckHook ];
-  disabledTests = [ "quart" "fastapi" ];
-  disabledTestPaths = [ "tests/test_fastapi.py" ];
+  checkInputs = [ wheel flask /*quart*/ sanic fastapi uvicorn requests pytestCheckHook ];
+  disabledTests = [
+    "quart" # Quart is not packaged for Nixpkgs
+  ] ++ lib.optionals (pythonAtLeast "3.9") [
+    "test_api_example" # network access
```
Comment (dotlambda) [inline pkgs/development/python-modules/json-logging/default.nix:37]: This is really strange.

Comment (dotlambda) [reply pkgs/development/python-modules/json-logging/default.nix:37]: Did urllib3 somehow strange in Python 3.9 in a way such that local networking doesn't work in the sandbox anymore?
```
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7ffff50a5f70>: Failed to establish a new connection: [Errno 111] Connection refused
```
cc @FRidh @jonringer

File: pkgs/development/python-modules/json-logging/default.nix [hunk 2]
```diff
@@ -1,5 +1,6 @@
 { lib
 , buildPythonPackage
+, pythonAtLeast
```
Comment (AluisioASG) [inline pkgs/development/python-modules/json-logging/default.nix:3]:
```suggestion
```

**Review by AluisioASG — CHANGES_REQUESTED** [review]
LGTM, save for the leftover `pythonAtLeast`.

**Review by AluisioASG — APPROVED** [review]

**Review by AluisioASG — APPROVED** [review]

**Review by costrouc — APPROVED** [review]

**Pull request merged**